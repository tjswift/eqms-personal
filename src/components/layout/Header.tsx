import * as React from 'react';
import SettingsMenu from './SettingsMenu';
import { Colors, Navbar, Alignment, Popover, Button, ButtonGroup, Position} from '@blueprintjs/core';

interface HeaderProps {
    title: string,
    welcomeMessage: string
}

const Header: React.SFC<HeaderProps> = ({ title, welcomeMessage }) => (

    <Navbar style={{ color: Colors.WHITE, background: Colors.BLUE2 }}>
        <Navbar.Group align={Alignment.LEFT} className="title">
            <Navbar.Heading>{title}</Navbar.Heading>
            <Navbar.Divider style={{ background: Colors.WHITE }}/>
        </Navbar.Group>
        <Navbar.Group className="navigation">
            <ButtonGroup minimal={true}>
                <a href="/"><Button title="Dashboard" text="Dashboard" icon="dashboard" intent="primary" style={{ color: Colors.WHITE }}/></a>
                <a href="/lessons"><Button title="Lessons Learned" text="Lessons Learned" icon="clipboard" intent="primary" style={{ color: Colors.WHITE }} /></a>
            </ButtonGroup>
        </Navbar.Group>
        <Navbar.Group align={Alignment.RIGHT} className="userSettings">
            <ButtonGroup minimal={true} className="SettingsGroup">       
                <Button className="userButton" title="UserIcon" text={welcomeMessage} rightIcon="user" intent="primary" style={{ color: Colors.WHITE }} />
                <Popover content={ <SettingsMenu/> }
                    position={Position.BOTTOM}>
                    <Button title="Settings" icon="cog" intent="primary" style={{ color: Colors.WHITE }} />
                </Popover>
            </ButtonGroup>
        </Navbar.Group>
    </Navbar>
)

export default Header
