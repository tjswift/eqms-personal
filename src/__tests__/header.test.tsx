/** Current tests include
 *The header renders (it’s there, it exists)
 * The elements are there and are in the correct order/have the correct alignment
 * The user message is rendered correctly
 **/
import React from 'react';
import Header from '../components/layout/Header'
import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer' 

configure({ adapter: new Adapter() });

describe('<Header />', () => {
    const title = 'EQMS Application';
    //todo: dynamic welcome message
    const welcomeMessageText = 'Welcome Firstname';

    const headerObj = < Header title={title} welcomeMessage={welcomeMessageText} />;
    const menuProps = mount(headerObj);
    const menuWrapper = shallow(headerObj);


    it('renders everything correct (snapshot)', () => {
        const component = renderer.create(<Header title={title} welcomeMessage={welcomeMessageText} />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('renders', () => {
        expect(menuWrapper.find('Navbar')).toBeTruthy();
    });

    it('renders title first', () => {
        expect(menuWrapper.childAt(0).hasClass("title")).toBe(true);
    });

    it('renders navigation second', () => {
        expect(menuWrapper.childAt(1).hasClass("navigation")).toBe(true);
    });

    it('renders user/settings third', () => {
        expect(menuWrapper.childAt(2).hasClass("userSettings")).toBe(true);
    });

    it('renders the correct user message', () => {
        expect(menuProps.props().welcomeMessage).toEqual(welcomeMessageText);
    });

    
});
