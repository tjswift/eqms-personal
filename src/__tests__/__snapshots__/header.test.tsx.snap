// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshots 1`] = `
<div
  className="bp3-navbar"
>
  <div
    className="bp3-navbar-group bp3-align-left title"
  >
    <div
      className="bp3-navbar-heading"
    >
      EQMS Application
    </div>
    <div
      className="bp3-navbar-divider"
    />
  </div>
  <div
    className="bp3-navbar-group bp3-align-left navigation"
  >
    <div
      className="bp3-button-group bp3-minimal"
    >
      <a
        href="/"
      >
        <button
          className="bp3-button"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          title="Dashboard"
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-dashboard"
            icon="dashboard"
          >
            <svg
              data-icon="dashboard"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                dashboard
              </desc>
              <path
                d="M5 4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zM4 7c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm4-2c.55 0 1-.45 1-1s-.45-1-1-1-1 .45-1 1 .45 1 1 1zm-2 6c0 1.1.9 2 2 2s2-.9 2-2c0-.53-2-5-2-5s-2 4.47-2 5zM8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm4-9c0-.55-.45-1-1-1s-1 .45-1 1 .45 1 1 1 1-.45 1-1zm0 2c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                fillRule="evenodd"
              />
            </svg>
          </span>
          <span
            className="bp3-button-text"
          >
            Dashboard
          </span>
        </button>
      </a>
      <a
        href="/lessons"
      >
        <button
          className="bp3-button"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          title="Lessons Learned"
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-clipboard"
            icon="clipboard"
          >
            <svg
              data-icon="clipboard"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                clipboard
              </desc>
              <path
                d="M11 2c0-.55-.45-1-1-1h.22C9.88.4 9.24 0 8.5 0S7.12.4 6.78 1H7c-.55 0-1 .45-1 1v1h5V2zm2 0h-1v2H5V2H4c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h9c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1z"
                fillRule="evenodd"
              />
            </svg>
          </span>
          <span
            className="bp3-button-text"
          >
            Lessons Learned
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    className="bp3-navbar-group bp3-align-right userSettings"
  >
    <div
      className="bp3-button-group bp3-minimal SettingsGroup"
    >
      <button
        className="bp3-button"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        title="UserIcon"
        type="button"
      >
        <span
          className="bp3-button-text"
        >
          Welcome Firstname
        </span>
        <span
          className="bp3-icon bp3-icon-user"
          icon="user"
        >
          <svg
            data-icon="user"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              user
            </desc>
            <path
              d="M7.99-.01A7.998 7.998 0 0 0 .03 8.77c.01.09.03.18.04.28.02.15.04.31.07.47.02.11.05.22.08.34.03.13.06.26.1.38.04.12.08.25.12.37.04.11.08.21.12.32a6.583 6.583 0 0 0 .3.65c.07.14.14.27.22.4.04.07.08.13.12.2l.27.42.1.13a7.973 7.973 0 0 0 3.83 2.82c.03.01.05.02.07.03.37.12.75.22 1.14.29l.2.03c.39.06.79.1 1.2.1s.81-.04 1.2-.1l.2-.03c.39-.07.77-.16 1.14-.29.03-.01.05-.02.07-.03a8.037 8.037 0 0 0 3.83-2.82c.03-.04.06-.08.09-.13.1-.14.19-.28.28-.42.04-.07.08-.13.12-.2.08-.13.15-.27.22-.41.04-.08.08-.17.12-.26.06-.13.11-.26.17-.39.04-.1.08-.21.12-.32.04-.12.08-.24.12-.37.04-.13.07-.25.1-.38.03-.11.06-.22.08-.34.03-.16.05-.31.07-.47.01-.09.03-.18.04-.28.02-.26.04-.51.04-.78-.03-4.41-3.61-7.99-8.03-7.99zm0 14.4c-1.98 0-3.75-.9-4.92-2.31.67-.36 1.49-.66 2.14-.95 1.16-.52 1.04-.84 1.08-1.27.01-.06.01-.11.01-.17-.41-.36-.74-.86-.96-1.44v-.01c0-.01-.01-.02-.01-.02-.05-.13-.09-.26-.12-.39-.28-.05-.44-.35-.5-.63-.06-.11-.18-.38-.15-.69.04-.41.2-.59.38-.67v-.06c0-.51.05-1.24.14-1.72.02-.13.05-.26.09-.39.17-.59.53-1.12 1.01-1.49.49-.38 1.19-.59 1.82-.59.62 0 1.32.2 1.82.59.48.37.84.9 1.01 1.49.04.13.07.26.09.4.09.48.14 1.21.14 1.72v.07c.18.08.33.26.37.66.03.31-.1.58-.16.69-.06.27-.21.57-.48.62-.03.13-.07.26-.12.38 0 .01-.01.04-.01.04-.21.57-.54 1.06-.94 1.42 0 .06.01.13.01.19.04.43-.12.75 1.05 1.27.65.29 1.47.6 2.14.95a6.415 6.415 0 0 1-4.93 2.31z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
      <span
        className="bp3-popover-wrapper"
      >
        <span
          className="bp3-popover-target"
          onClick={[Function]}
        >
          <button
            className="bp3-button"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            title="Settings"
            type="button"
          >
            <span
              className="bp3-icon bp3-icon-cog"
              icon="cog"
            >
              <svg
                data-icon="cog"
                height={16}
                viewBox="0 0 16 16"
                width={16}
              >
                <desc>
                  cog
                </desc>
                <path
                  d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 0 0-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 0 0-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </div>
  </div>
</div>
`;
